### DISCUSSION TITLE
MacOS ctrl-w Custom keymap is way slower than default

### DISCUSSION THREAD

**Discussion by lakardion** [zed-industries/zed#8274] [Feature Requests]
I'm used to using CTRL for the basic tasks such as saving and closing windows.

`keymap.json`
```json
[
  {
    "bindings": {
      "ctrl-s": "workspace::Save",
      "ctrl-k ctrl-s": "workspace::SaveAll",
      "ctrl-w": "pane::CloseActiveItem",
      "ctrl-k i": "editor::Hover"
    }
  }
]
```
Whenever I try to close a window with `ctrl-w` it takes like 1 second to actually do it. 

Whereas if I do with the default keybinding (`cmd-w`) the action happens instantly.

I thought there might be some threshold time for listening for another keybind such as the many cmd-k ones. But I checked the default keymap and there is nothing with ctrl-w

Is there something I'm missing that could be causing this?
I'm in MacOS Zed 0.123.3

**Comment by Moshyfawn** [comment]
I can't seem to reproduce the described behaviour in either Preview 0.124.1 or Stable 0.123.3. Are there any logs or anything that Zed might have generated?

**Reply by l-7-l** [reply]
same here on linux.
editor: 
  version: Zed Preview 0.144.2
  code: 
![image](https://github.com/user-attachments/assets/6283f71d-8714-479a-8c13-c8a6aaf55b0a)

os:
```
                   -`                    l-7-l@bb8 
                  .o+`                   --------- 
                 `ooo/                   OS: Arch Linux x86_64 
                `+oooo:                  Host: Vivobook_ASUSLaptop M7600QE_M7600QE 1.0 
               `+oooooo:                 Kernel: 6.9.7-zen1-1-zen 
               -+oooooo+:                Uptime: 12 hours, 57 mins 
             `/:-:++oooo+:               Packages: 1203 (pacman) 
            `/++++/+++++++:              Shell: zsh 5.9 
           `/++++++++++++++:             Resolution: 3840x2400 
          `/+++ooooooooooooo/`           DE: GNOME 46.3.1 
         ./ooosssso++osssssso+`          WM: Mutter 
        .oossssso-````/ossssss+`         WM Theme: Adwaita 
       -osssssso.      :ssssssso.        Theme: Adwaita [GTK2/3] 
      :osssssss/        osssso+++.       Icons: Adwaita [GTK2/3] 
     /ossssssss/        +ssssooo/-       Terminal: gjs 
   `/ossssso+/:-        -:/+osssso+-     CPU: AMD Ryzen 9 5900HX with Radeon Graphics (16) @ 4.680GHz 
  `+sso+:-`                 `.-/+oso:    GPU: NVIDIA GeForce RTX 3050 Ti Mobile 
 `++:.                           `-/+/   GPU: AMD ATI Radeon Vega Series / Radeon Vega Mobile Series 
 .`                                 `/   Memory: 9315MiB / 15389MiB 

```
[录屏 2024-07-16 13-47-01.webm](https://github.com/user-attachments/assets/dbebb797-4909-44e5-a220-981fa4510f8b)

**Reply by CharlesChen0823** [reply]
might cause by `"vim_mode" : "true"`.

**Reply by tepavcevic** [reply]
Vim mode uses `ctrl+w` as modifier key or leader or whatever you want to call it, so it waits for the complete command before defaulting to closing the tab. You can see the keymap by searching for `vim: open default keymap` in control panel.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
